### PR TITLE
MGMT-4477 Pre-flight host requirements API

### PR DESCRIPTION
**Pull request by jakub-dzon**
This PR introduces new API for providing pre-flight requirements in context of a cluster (but not hosts).
The new endpoint provides requirements for both OCP and supported operators.

Handler for the API will be implemented separately.

### CODE REVIEW

**Comment by omertuc** [comment]
Please include the PR description in the commit message

File: swagger.yaml [hunk 1]
```diff
@@ -3249,6 +3290,54 @@ definitions:
...
 
+  preflight-requirements:
+    type: object
+    properties:
+      operators:
+        description: Preflight operators requirements
+        type: array
+        items:
+          $ref: '#/definitions/operator-requirements'
+      ocp:
```
Comment (omertuc) [inline swagger.yaml:3301]: Can we use a more indicative name? Maybe something that includes the word `hardware`?

File: swagger.yaml [hunk 2]
```diff
@@ -3249,6 +3290,54 @@ definitions:
...
+    type: object
+    properties:
+      operator_name:
+        description: Name of the operator
+        type: string
+      dependencies:
+        description: List of other operators that are required to be installed
+        type: array
+        items:
+          type: string
```
Comment (omertuc) [inline swagger.yaml:3315]: What's the exact format of this string? Is it the operator name or some ID/URL? maybe add an example / more detailed description of how operators are identified

File: swagger.yaml [hunk 3]
```diff
@@ -3249,6 +3290,54 @@ definitions:
...
+      ocp:
+        description: Preflight OCP requirements
+        $ref: '#/definitions/host-type-requirements-wrapper'
+
+  operator-requirements:
+    type: object
+    properties:
+      operator_name:
+        description: Name of the operator
+        type: string
```
Comment (omertuc) [inline swagger.yaml:3310]: Same as https://github.com/openshift/assisted-service/pull/1508/files#r613100505

File: swagger.yaml (hunk 1 above)
Comment (jakub-dzon) [reply swagger.yaml:3301]: Good idea.

**Comment by mareklibra** [comment]
Looks good from the UI perspective

**Review by avishayt — APPROVED** [review]

**Pull request merged**